lib/sshutils/x11/display.go

```diff
@@ -227,5 +227,7 @@ func ParseDisplay(displayString string) (Display, error) {
 }
 
 func x11SockDir() string {
-	return filepath.Join(os.TempDir(), x11SocketDirName)
+	// We use "/tmp" instead of "os.TempDir" because X11
+	// is not OS aware and uses "/tmp" regardless of OS.
+	return filepath.Join("/tmp", x11SocketDirName)
```

Comment:
Even on Windows?

Reply:
I'm not sure, I've tested with `Xming`, `MobaXTerm`, and `VcXsrv` on Windows, which all use tcp sockets instead of unix sockets, so those won't be affected. I don't know of a Windows XServer that uses a unix socket, but my current assumption is that `/tmp/.X11-unix` is a hard coded value in the X Window System. A $DISPLAY like `unix:n` or `:n` always gets translated to `/tmp/.X11-unix/Xn` by X Programs from what I have tested.

Reply:
I don't think `/tmp` is ever going to work on windows, could we return an error?

Reply:
With MobaXterm, Cgywin, and other custom unix-like windows terminals, `/tmp` can be used this way. In fact, given the X11 protocol, I believe this is the only way for any OS to support unix sockets for their X Server.

I actually just tested out Cgywin/X because I found out that it uses unix sockets for its X Server implementation (`/tmp/.X11-unix/X0`), so I believe the current implementation is correct.

Reply:
hmm, TIL, thanks